generate/avro.go

```diff
@@ -0,0 +1,304 @@
+package generate
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+const (
+	avroHeader           = `{"name":%q,"type":"record","fields":[{"name":"AvroWriteTime","doc":"The timestamp when this avro data is written. Useful for identifying the newest row of data sharing keys.","type":"long","logicalType":"timestamp-millis"},{"name":"AvroDeleted","doc":"This is set to true when the Avro data is recording a delete in the source data.","default":false,"type":"boolean"},`
+	embeddedStructHeader = `{"name":%q,"type":"record","fields":[`
+)
+
+type avroConfig struct {
+	dir       string
+	namespace string
+	writer    io.Writer
+}
+
+// buildAvroSerializationFunctions will run gogen-avro on the given schema to build a serialization and deserialization
+// code for the schema.
+func buildAvroSerializationFunctions(outputDir string) error {
+	// TODO buildAvroSerializationFunctions - this should take the schema and run gogen-avro
```

Comment:
Is this something that will be done in another ticket/PR?

Reply:
Yes another PR or two is on its way. I think there is a 3 step process for the Avro code generation, step 1 is the Avro schema, step 2 is to make the Avro serialization/deserialization code, step 3 is to add helpers that make writing out Avro files easy for each data type.